src/flint/fq_nmod.jl

```diff
@@ -102,10 +102,7 @@ is_unit(a::fqPolyRepFieldElem) = ccall((:fq_nmod_is_invertible, libflint), Bool,
                      (Ref{fqPolyRepFieldElem}, Ref{fqPolyRepField}), a, a.parent)
 
 function characteristic(a::fqPolyRepField)
-   d = ZZRingElem()
```

Comment:
This would need a `GC.@preserve a begin` block. Similar for the other change. You for some other examples in the code to check that I got that syntax right.

Reply:
Sorry, but I have no knowledge of this: So even a simple access to a primitive type in the element `a` would not guarantee conservation of `a` before and during this function to the GC?

Remind me again, is the `ccall` some type of hint to the GC to preserve `a`? But I feel like I have seen `GC.@preserve` around `ccall`s as well in Nemo...

Reply:
> Sorry, but I have no knowledge of this: So even a simple access to a primitive type in the element `a` would not guarantee conservation of `a` before and during this function to the GC?

Yes. The call
```
z = f(a.n)
```
is equivalent to
```
m = a.n 
z = f(m)
```
but the GC might now run before the `f(m)` call.

> Remind me again, is the `ccall` some type of hint to the GC to preserve `a`? But I feel like I have seen `GC.@preserve` around `ccall`s as well in Nemo...

Yes, but for exactly the same reason. There are some `@preserve` calls around `*_mat_entry_ptr`, because the returned pointer does not know about the matrix and thus the matrix need to be kept alive to keep everyone happy.

Reply:
Ah, I see. Thank you! Will change later!